spec/Mapper/DonorMapperSpec.php

```diff
@@ -93,4 +93,20 @@ function it_can_delete_donor($collection, $donorSchema, Donor $donor, Expression
 
         $this->delete($donor);
     }
+
+    function it_can_find_report_existing_donor(
+        $collection
+    ) {
+        $collection->has('foobar')->willReturn(true);
+
+        $this->shouldHaveKey('foobar');
```

Comment:
We are testing the hasKey() method. `shouldHaveKey` is for iterable objects from what I understand.

http://www.phpspec.net/en/stable/cookbook/matchers.html#iterablekey-matcher

Try

```php
$this->hasKey('foobar')->shouldReturn(true);
```